packages/dd-trace/src/appsec/reporter.js

```diff
@@ -199,19 +200,40 @@ function finishRequest (req, res) {
   incrementWafRequestsMetric(req)
 
   // collect some headers even when no attack is detected
-  rootSpan.addTags(filterHeaders(req.headers, IDENTIFICATION_HEADERS_MAP))
+  const mandatoryTags = filterHeaders(req.headers, MANDATORY_REQUEST_HEADERS_MAP)
+  const ua = mandatoryTags['http.request.headers.user-agent']
+  if (ua) {
+    mandatoryTags['http.useragent'] = ua
+  }
+  rootSpan.addTags(mandatoryTags)
 
-  if (!rootSpan.context()._tags['appsec.event']) return
+  const tags = rootSpan.context()._tags
+  if (!shouldTrackHeaders(tags)) return
 
   const newTags = filterHeaders(res.getHeaders(), RESPONSE_HEADERS_MAP)
+  Object.assign(newTags, filterHeaders(req.headers, REQUEST_HEADERS_MAP))
 
-  if (req.route && typeof req.route.path === 'string') {
+  const appsecEvent = tags['appsec.event'] === 'true'
+  if (typeof req.route?.path === 'string' && appsecEvent) {
     newTags['http.endpoint'] = req.route.path
   }
```

Comment:
why you removing my newlines ?
```suggestion
  }

```